### PR TITLE
Allow a value of 0 with :vent-at

### PR DESCRIPTION
**Pull request by lizmat**
Hopefully fixing #5358

### CODE REVIEW

**Comment by lizmat** [comment]
@amano-kenji @AlexDaniel could you check this out the coming days so that it can make the 2023.10 release due this Thursday?

File: src/core.c/Supply-coercers.rakumod [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# continued from src/core.c/Supply-factories.rakumod
+# continued from src/core.c/Supply-factories.pm6
```
Comment (2colours) [inline src/core.c/Supply-coercers.rakumod:1]: The extension got reverted here

**Comment by amano-kenji** [comment]
I will test this next morning.

**Comment by amano-kenji** [comment]
I tried to test it. I built it with
```
perl Configure.pl --gen-moar --gen-nqp --backend=moar
```
But, I can't find `rakudo` executable anywhere.

When I call
```
multi method throttle(Supply:D:
      Int()  $elems,
      Callable:D $process,
      Real() $delay = 0,
      :$scheduler   = $*SCHEDULER,
      :$control,
      :$status,
      :$bleed,
      :$vent-at,
    )
```
I want only one process to be executed at a time, and while the process is being executed, I want the supply to drop all incoming messages. Does `vent-at(0)` do that?

**Comment by lizmat** [comment]
That's the idea.  If you also specify `:bleed` of course?

**Comment by amano-kenji** [comment]
If I also specify `:bleed`, I want the supply to send everything to `bleed` which will get rid of excess.

**Comment by amano-kenji** [comment]
If that works properly as expected, then I think it should be okay to merge it without us.

**Comment by lizmat** [comment]
> If that works properly as expected, then I think it should be okay to merge it without us.

That's it: I didn't test it, as I don't really have an easy test case.

Anyway, it is too late for the 2023.10 release, which will be today.

**Comment by amano-kenji** [comment]
How can I test this without installing rakudo?

**Comment by ab5tract** [comment]
@amano-kenji You have to use this git repo checked out to the branch of this PR (`lizmat-vet-0`). First you run `perl Configure.pl --gen-moar --gen-nqp --backends=moar` and _then_ run `make`. This will create executables in the directory. No installation necessary.

**Comment by amano-kenji** [comment]
I got `rakudo-m` executable which seemed to be what I wanted.

I executed this.

```
my $supplier = Supplier.new;

my $chan = $supplier.Supply.throttle(1, -> $a { sleep 5; $a; }, :bleed(Supplier.new), :vent-at(0)).Channel;

start {
  for [1..*] {
    sleep 0.1;
    $supplier.emit($_);
  }
}

react {
  whenever $chan {
    say $_.result;
  }
}
```

The output was

```
1
51
101
151
...
```

**Comment by ab5tract** [comment]
FWIW I was unable to achieve the behavior that @amano-kenji reported when using this branch on macOS 14.

**Comment by ab5tract** [comment]
I've rebased this against current `main` and still unfortunately do not get the expected results:

```
longwalker@mirabilis ~/c/r/z/rakudo ((6435e28b))
> ./rakudo-m ~/code/raku/scripts/ventatzeroraku/vent-at-zero.raku
1
49
97
145
194
242
291
```

**Pull request merged**